src/python/pants/backend/native/subsystems/native_build_step_settings_base.py

```diff
@@ -4,37 +4,114 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+from pants.backend.native.subsystems.native_build_settings import NativeBuildSettings
 from pants.backend.native.subsystems.utils.mirrored_target_option_mixin import \
   MirroredTargetOptionMixin
 from pants.subsystem.subsystem import Subsystem
+from pants.util.memo import memoized_property
 
 
 class NativeBuildStepSettingsBase(Subsystem, MirroredTargetOptionMixin):
 
+  options_scope = 'native-build-step-settings-base'
+
   mirrored_option_to_kwarg_map = {
     'fatal_warnings': 'fatal_warnings',
+    'compiler_option_sets': 'compiler_option_sets',
   }
 
+  @classmethod
+  def subsystem_dependencies(cls):
+    return super(NativeBuildStepSettingsBase, cls).subsystem_dependencies() + (
+      NativeBuildSettings.scoped(cls),
+    )
+
+  @memoized_property
+  def _native_build_settings(self):
+    return NativeBuildSettings.scoped_instance(self)
+
   @classmethod
   def register_options(cls, register):
     super(NativeBuildStepSettingsBase, cls).register_options(register)
 
-    # TODO: implement compiler_option_sets as an interface to platform/host-specific optimization
-    # flags!
+    # TODO: Deprecate `--fatal-warnings` in a follow-up revision.
     register('--fatal-warnings', type=bool, default=True, fingerprint=True, advanced=True,
              help='The default for the "fatal_warnings" argument for targets of this language.')
+    register('--compiler-option-sets-enabled-args', advanced=True, type=dict, fingerprint=True,
```

Comment:
These are defined in 3 places within the backend now... was the intention for these all to be on the scoped subsystem? Or to be inherited?